node_modules/col-categories/lib/api.js

```diff
@@ -0,0 +1,292 @@
+    if (err) {
+      log.error({'err': err, 'course': ctx.course}, 'Failed to get the categories in the current course');
+      return callback({'code': 500, 'msg': err.message});
+    }
+
+    return callback(null, results[0]);
+  });
+};
+
+/**
+ * Get a set of categories by their id
+ *
+ * @param  {Context}        ctx                         Standard context containing the current user and the current course
+ * @param  {Number[]}       ids                         The ids of the categories that should be retrieved
+ * @param  {Function}       callback                    Standard callback function
+ * @param  {Object}         callback.err                An error that occurred, if any
+ * @param  {Category[]}     callback.categories         The retrieved categories
+ */
+var getCategoriesById = module.exports.getCategoriesById = function(ctx, ids, callback) {
+  // Return immediately if no category ids have been provided
+  if (!ids || ids.length === 0) {
+    return callback(null, []);
+  }
+
+  // Parameter validation
+  var validationSchema = Joi.object().keys({
+    'ids': Joi.array().unique().items(Joi.number())
+  });
+
+  var validationResult = Joi.validate({
+    'ids': ids
+  }, validationSchema);
+
+  if (validationResult.error) {
+    return callback({'code': 400, 'msg': validationResult.error.details[0].message});
+  }
+
+  // Get the categories from the DB
+  var options = {
+    'where': {
+      'course_id': ctx.course.id,
+      'id': ids
+    }
+  };
+
+  DB.Category.findAll(options).complete(function(err, categories) {
+    if (err) {
+      log.error({'err': err, 'ids': ids}, 'Failed to get a set of categories');
+      return callback({'code': 500, 'msg': err.message});
+    } else if (categories.length !== ids.length) {
+      log.error({'err': err, 'ids': ids}, 'Could not retrieve all requested categories');
+      return callback({'code': 404, 'msg': 'Could not retrieve all requested categories'});
+    }
+
+    return callback(null, categories);
+  });
+};
+
+/**
+ * Create a new category
+ *
+ * @param  {Context}        ctx                         Standard context containing the current user and the current course
+ * @param  {String}         title                       The name of the category
+ * @param  {Number}         [canvasAssignmentId]        The id of the assignment to which this category is linked
+ * @param  {Function}       callback                    Standard callback function
+ * @param  {Object}         callback.err                An error that occurred, if any
+ * @param  {Category}       callback.category           The created category
+ */
+var createCategory = module.exports.createCategory = function(ctx, title, canvasAssignmentId, callback) {
+  // Parameter validation
+  var validationSchema = Joi.object().keys({
+    'title': Joi.string().max(255).required(),
+    'canvasAssignmentId': Joi.number().optional()
```

Comment:
It's not possible to set it through the REST APIs, but once we'll need it to create a new category when a new assignment is set up.